pkg/policy/api/rule_validation.go

```diff
@@ -641,15 +641,20 @@ func (pr *PortDenyRule) sanitize() error {
 
 func (pp *PortProtocol) sanitize(hasDNSRules bool) (isZero bool, err error) {
 	if pp.Port == "" {
-		return isZero, errors.New("Port must be specified")
+		if !option.Config.EnableExtendedIPProtocols {
+			return isZero, errors.New("port must be specified")
+		}
 	}
 
 	// Port names are formatted as IANA Service Names.  This means that
 	// some legal numeric literals are no longer considered numbers, e.g,
 	// 0x10 is now considered a name rather than number 16.
 	if iana.IsSvcName(pp.Port) {
 		pp.Port = strings.ToLower(pp.Port) // Normalize for case insensitive comparison
-	} else {
+	} else if pp.Port != "" {
+		if pp.Port != "0" && (pp.Protocol == ProtoVRRP || pp.Protocol == ProtoIGMP) {
+			return isZero, errors.New("port must be empty or 0")
+		}
```

Comment:
When `EnableExtendedIPProtocols` is true, an unspecified `Port` with UDP or TCP `Protocol` is allowed when it used to be rejected before this commit. Is that expected?

Reply:
Yes, see - https://github.com/cilium/cilium/pull/39872#issuecomment-2977754154 and https://github.com/cilium/cilium/pull/39872#discussion_r2144511283. 
I'm not sure if this warrants a note in the upgrade guide though.

Reply:
Sure, I'll add as a follow-up.